js/ui/grid_core/ui.grid_core.search.js

```diff
@@ -343,16 +343,30 @@ module.exports = {
                 },
 
                 _updateCell: function($cell, parameters) {
-                    var that = this,
-                        column = parameters.column,
+                    var column = parameters.column,
                         dataType = column.lookup && column.lookup.dataType || column.dataType,
                         isEquals = dataType !== "string";
 
                     if(allowSearch(column)) {
-                        that._highlightSearchText($cell, isEquals, column);
+                        if(this.option("templatesRenderAsynchronously")) {
+                            if(!this._searchParams) {
+                                this._searchParams = [];
+
+                                clearTimeout(this._highlightTimer);
+
+                                this._highlightTimer = setTimeout(function() {
+                                    this._searchParams.forEach(function(params) {
+                                        this._highlightSearchText.apply(this, params);
+                                    }.bind(this));
```

Comment:
The this._searchParams array should be cleared at the end of timeout. Otherwise, they count will be increased on each render.